### PR TITLE
Avoid overflowing the stack in consume_until_end_of_block.

### PR DESCRIPTION
**Pull request by emilio**
Fixes: https://bugzilla.mozilla.org/show_bug.cgi?id=1323705

r? @SimonSapin

---
This change is [Reviewable](https://reviewable.io/reviews/servo/rust-cssparser/111)

### CODE REVIEW

File: src/parser.rs [hunk 1]
```diff
@@ -691,15 +691,22 @@ impl<'i, 't> Parser<'i, 't> {
...
 
-/// Return value indicates whether the end of the input was reached.
 fn consume_until_end_of_block(block_type: BlockType, tokenizer: &mut Tokenizer) {
+    let mut stack = vec![block_type];
+
     // FIXME: have a special-purpose tokenizer method for this that does less work.
     while let Ok(ref token) = tokenizer.next() {
-        if BlockType::closing(token) == Some(block_type) {
-            return
+        if let Some(b) = BlockType::closing(token){
```
Comment (nox) [inline src/parser.rs]: Space before brace.

**Comment by emilio** [comment]
Done

**Comment by SimonSapin** [comment]
r=me with a semver-compatible version bump.

**Pull request merged**